src/Chapter1.hs

```diff
@@ -542,7 +544,14 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+mid :: Ord a => a -> a -> a -> a
+mid x y z
+    | x < y && y < z = y
+    | x < z && z < y = z
+    | y < x && x < z = x
+    | y < z && z < x = z
+    | z < x && x < y = x
+    | otherwise = y
```

Comment:
That is right! 👌🏼 

You can notice that the first guard is actually covered by the last one `otherwise = y`, so you can remove it 🙂

Reply:
That's a great point, I spent a lot of time testing all 6 cases and once I got it working I was too burnt to further optimize. What you say makes sense though, I really only need to find outcomes for 2 of the 3 variables, then everything else will be the third variable. Still wrapping my head around the power of otherwise!